File: dog.rb
class Dog

	def initialize(args)
		@name = args.fetch(:name, "Puddles")
		@breed = args.fetch(:breed, "Corgy")
		@age = args.fetch(:age, "1")
	end

  def bark
    puts "Woof woof"
  end
  
end

dog = Dog.new({name: "Chancy", breed: "Dalmation", age: 11})
dog.bark

File: dog.js
function Dog(args){
	this.name = args["name"] || "Potato monster";
	this.breed = args["breed"] || "Poodle";
	this.age = args.age || 1;

	// Don't do this
	// this.bark = function(){
	// 	console.log("bow wow");
	// }
}



Dog.prototype.bark = function(){
	console.log("Bow wow");
}

var dog = new Dog({name: "Skittles"});
console.log(dog);
dog.bark();